src/core/model/reducer.ts

```diff
@@ -92,11 +75,21 @@ export const uiSchemaReducer = (
 
 export const combinedReducer = (state: EditorState, action: CombinedAction) => {
   switch (action.type) {
+    case SET_SCHEMA:
+      return buildAndLinkUISchema(
+        buildSchemaTree(action.schema),
+        state.uiSchema
+      );
+    case SET_UISCHEMA:
+      return buildAndLinkUISchema(
+        state.schema ? cleanLinkedElements(state.schema) : state.schema,
```

Comment:
wouldn't it be a bit cleaner to just let `cleanLinkedElements` handle the undefined?